### ISSUE TITLE
`__typename` hardcoded in decoder for union types

### ISSUE THREAD

**Issue by arian** [harmboschloo/graphql-to-elm#8]
Say this schema:

```graphql
type Query {
  foo: SomeUnionType
}
union SomeUnionType = Connected | NotConnected
type Connected {
  id: ID
}
type NotConnected {
  id: ID
}
```

Then the query

```graphql
{ foo { __typename } }
```

generates a decoder

```elm
        (Json.Decode.field "__typename" (GraphQL.Helpers.Decode.constantString "Connected"))
```

however in this case, it could actually be either `Connected` or `NotConnected`!

The workaround is to use a query with all cases specified:

```graphql
{ foo {
  ... on Connected { __ typename } 
  ... on NotConnected { __typename }
 } }
```

**Comment by arian** [comment]
Probably this would need to be a bit expanded: https://github.com/harmboschloo/graphql-to-elm/blob/master/src/queries/elmIntel.ts#L494-L500

**Comment by harmboschloo** [comment]
Good catch! 😄

**Comment by harmboschloo** [comment]
fixed by 2130d9e25f5269e6cf21cd96b099df2f022a16c2
